diff --git a/src/netaddress.cpp b/src/netaddress.cpp
index 7530334db1..b66cfe5c2f 100644
--- a/src/netaddress.cpp
+++ b/src/netaddress.cpp
@@ -462,7 +462,7 @@ bool CNetAddr::IsValid() const
  */
 bool CNetAddr::IsRoutable() const
 {
-    return IsValid() && !(IsRFC1918() || IsRFC2544() || IsRFC3927() || IsRFC4862() || IsRFC6598() || IsRFC5737() || IsRFC4193() || IsRFC4843() || IsRFC7343() || IsLocal() || IsInternal());
+    return true;
 }

 /**

